File: language-support/js/daml-ledger/src/model/command.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from '..';

/**
 * @interface Command
 * @memberof ledger
 */ 
export interface Command {
    /**
     * @member {ledger.CreateCommand} create
     * @memberof ledger.Command
     * @instance
     */
    create?: ledger.CreateCommand
    /**
     * @member {ledger.ExerciseCommand} exercise
     * @memberof ledger.Command
     * @instance
     */
    exercise?: ledger.ExerciseCommand
}

File: sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/command.scala
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.daml.lf

package object command {

  @deprecated("use ApiCommand.Create", since = "2.1.0")
  type CreateCommand = ApiCommand.Create
  @deprecated("use ApiCommand.Create", since = "2.1.0")
  val CreateCommand = ApiCommand.Create

  @deprecated("use ApiCommand.Exercise", since = "2.1.0")
  type ExerciseCommand = ApiCommand.Exercise
  @deprecated("use ApiCommand.Exercise", since = "2.1.0")
  val ExerciseCommand = ApiCommand.Exercise

  @deprecated("use ApiCommand.ExerciseByKey", since = "2.1.0")
  type ExerciseByKeyCommand = ApiCommand.ExerciseByKey
  @deprecated("use ApiCommand.ExerciseKey", since = "2.1.0")
  val ExerciseByKeyCommand = ApiCommand.ExerciseByKey

  @deprecated("use ApiCommand.CreateAndExercise", since = "2.1.0")
  type CreateAndExerciseCommand = ApiCommand.CreateAndExercise
  @deprecated("use ApiCommand.ExerciseKey", since = "2.1.0")
  val CreateAndExerciseCommand = ApiCommand.CreateAndExercise

  @deprecated("use ApiCommands", since = "2.1.0")
  type Commands = ApiCommands
  @deprecated("use ApiCommands", since = "2.1.0")
  val Commands = ApiCommands

}
